storage/bootstrap/bootstrapper/peers/source.go

```diff
@@ -117,5 +199,103 @@ func (s *peersSource) Read(
 
 	wg.Wait()
 
+	close(incrementalQueue)
+	incrementalWg.Wait()
+
+	if incremental {
+		// Now cache the incremental results
+		shards := make([]uint32, 0, len(shardsTimeRanges))
+		for shard := range shardsTimeRanges {
```

Comment:
It doesn't hurt to try and cache the shard indices, it simply scans the available filesets.  At most this would save a few milliseconds of disk seeks.  It doesn't return an error if the filesets are not there as it is best effort.